### PR TITLE
Working JDK10 version -- this checkpoint can be merged to the jdk10 branch

### PR DESCRIPTION
**Pull request by xythian**

### CODE REVIEW

File: yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ClosedPropertyAdapter.java [hunk 1]
```diff
@@ -45,6 +38,20 @@ public final AssignableValue property(BytecodeExpression target, String property
         return getPropertyValue(target, getProperty(propertyName).name);
     }
 
+    @Override
+    public BytecodeExpression property(BytecodeExpression target, String propertyName, BytecodeExpression defaultValue) {
+        try {
+            BytecodeExpression propertyValue = getPropertyValue(target, getProperty(propertyName).name);
+            if(propertyValue.getType().isNullable()) {
```
Comment (dsyzhu) [inline yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ClosedPropertyAdapter.java]: missing one space

File: yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ClosedPropertyAdapter.java [hunk 2]
```diff
@@ -45,6 +38,20 @@ public final AssignableValue property(BytecodeExpression target, String property
...
+                return new CoalesceExpression(propertyValue.getType().boxed(), propertyValue, defaultValue);
+            } else {
+                return propertyValue;
+            }
+        } catch(PropertyNotFoundException e) {
```
Comment (dsyzhu) [inline yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ClosedPropertyAdapter.java]: missing one space

File: yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/CoalesceExpression.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        this.primaryValue = primaryValue;
+        this.defaultValue = defaultValue;
+    }
+
+    @Override
+    public void generate(CodeEmitter code) {
+        AssignableValue val = code.allocate(primaryValue);
+        Label isNull = new Label();
+        Label isDone = new Label();
+        if(code.gotoIfNull(val, isNull)) {
```
Comment (dsyzhu) [inline yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/CoalesceExpression.java]: missing space

File: yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/DefaultTypeAdapter.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            @Override
+            public boolean supports(Class<?> clazzType) {
+                return Future.class.isAssignableFrom(clazzType);
+            }
+
+            @Override
+            public TypeWidget adapt(EngineValueTypeAdapter typeAdapter, Type type) {
+                TypeWidget valueType = typeAdapter.adapt(JVMTypes.getTypeArgument(type, 0));
+                Class<?> rawType = JVMTypes.getRawType(type);
+                if(ListenableFuture.class.isAssignableFrom(rawType)) {
```
Comment (dsyzhu) [inline yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/DefaultTypeAdapter.java]: missing space

File: yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/DefaultTypeAdapter.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                    return result;
+                }
+            }
+        }
+        return null;
+    }
+
+    private org.objectweb.asm.Type getPackageSafeRawType(Type type, Class<?> acceptableType) {
+        Class<?> clazz = JVMTypes.getRawType(type);
+        if(!Modifier.isPublic(clazz.getModifiers())) {
```
Comment (dsyzhu) [inline yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/DefaultTypeAdapter.java]: missing space

File: yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ExactInvocation.java [hunk 6]
```diff
@@ -4,25 +4,42 @@
...
 import java.util.Arrays;
 import java.util.List;
 
 public class ExactInvocation extends BytecodeInvocable {
+    public static GambitCreator.Invocable reflectInvoke(EngineValueTypeAdapter adapter, TypeLiteral<?> owner, Method method) {
+        TypeWidget ownerType = adapter.adapt(owner);
+        TypeWidget returnType = adapter.adapt(owner.getReturnType(method));
+        String methodName = method.getName();
+        List<TypeWidget> argTypes = Lists.newArrayListWithExpectedSize(method.getParameterCount());
+        for(TypeLiteral<?> argType : owner.getParameterTypes(method)) {
```
Comment (dsyzhu) [inline yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ExactInvocation.java]: missing space

File: yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ExactInvocation.java [hunk 7]
```diff
@@ -4,25 +4,42 @@
...
+            argTypes.add(adapter.adapt(argType));
+        }
+        int op = Opcodes.INVOKEVIRTUAL;
+        if(owner.getRawType().isInterface()) {
```
Comment (dsyzhu) [inline yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ExactInvocation.java]: missing space

File: yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ExactInvocation.java [hunk 8]
```diff
@@ -4,25 +4,42 @@
...
+            op = Opcodes.INVOKEINTERFACE;
+        }
+        if(Modifier.isStatic(method.getModifiers())) {
```
Comment (dsyzhu) [inline yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ExactInvocation.java]: missing space

Comment (xythian) [reply yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ExactInvocation.java]: Pushed an update that ran Code Format on all files -- there were many missing spaces even beyond all these.

**Comment by dsyzhu** [comment]
👍

**Pull request merged**